client/state/posts/selectors.js

```diff
@@ -34,23 +34,32 @@ export function getPost( state, globalId ) {
 }
 
 /**
- * Returns a normalized post object by its global ID.
+ * Returns a normalized post object by its global ID, or null if the post does
+ * not exist. A normalized post includes common transformations to prepare the
+ * post for display.
  *
- * @param  {Object} state    Global state tree
- * @param  {String} globalId Post global ID
- * @return {Object}          Post object
+ * @param  {Object}  state    Global state tree
+ * @param  {String}  globalId Post global ID
+ * @return {?Object}          Post object
  */
 export const getNormalizedPost = createSelector(
 	( () => {
-		// Cache normalize flow in immediately-invoked closure so to avoid
+		// Cache normalize flow in immediately-invoked closure to avoid
 		// regenerating same flow on each call to this selector
 		const normalize = flow( [
 			firstPassCanonicalImage,
 			decodeEntities,
 			stripHtml
 		] );
 
-		return ( state, globalId ) => normalize( cloneDeep( getPost( state, globalId ) ) );
+		return ( state, globalId ) => {
+			const post = getPost( state, globalId );
+			if ( ! post ) {
+				return null;
+			}
+
+			return normalize( cloneDeep( post ) );
```

Comment:
>but isn't this going to throw away all of the normalized posts when any one changes?

Yeah, it will. In fact, [the reducer for `state.posts.items`](https://github.com/Automattic/wp-calypso/blob/4b656271fd3c405da699475cccc39bcb19e3808b/client/state/posts/reducer.js#L39-L74) is pretty dumb and will return a new object even if none of the posts truly actually changed. We could probably optimize that, then perhaps change the `getDependents` argument (second argument) of `createSelector` here to more precisely target the post that'd change.